src/com/twu/biblioteca/MainMenu.java

```diff
@@ -32,36 +34,71 @@ public void showGoodbyeMessage() {
         System.out.println(goodBye);
     }
 
+    public void showBookCheckOutReferenceMessage() {
+        System.out.println(checkOutBookReference);
+    }
+
     public void showOptions() {
-        for(int i = 0; i < options.size(); i++) {
+        for (int i = 0; i < options.size(); i++) {
             System.out.println(options.get(i));
         }
         System.out.print(">");
     }
 
-    public void showList() {
-        for(int i = 0; i < books.size(); i++) {
-            System.out.println(books.get(i));
-        }
-    }
-
     public void manageOptions() {
         int counter = 0;
-        Scanner scanner = new Scanner(System.in);
+        Scanner optionsScanner = new Scanner(System.in);
 
         while (counter == 0) {
-            String userInput = scanner.next();
-            if(userInput.equals("1")) {
-                showList();
-                counter = 1;
-            }
-            else if(userInput.equals("2")) {
+            String userInput = optionsScanner.next();
+            if (userInput.equals("1")) {
+                showBookList();
+                showOptions();
+            } else if (userInput.equals("2")) {
+                showBookList();
+                manageBookList();
+            } else if (userInput.equals("3")) {
                 showGoodbyeMessage();
                 System.exit(0);
-            }
-            else {
+            } else {
                 showInvalidMessage();
             }
         }
     }
+
+    public void showBookList() {
+        for (int i = 0; i < this.books.size(); i++) {
+            Book book = this.books.get(i);
+            if (book.getCheckedOut() == false) {
+                int count = countCheckedOutBooks();
+                System.out.println(String.format(
+                        "%d. %s | %s | %d",
+                        (i + 1) - count, book.getTitle(), book.getAuthor(), book.getYearPublished()));
```

Comment:
The responsibility for formatting a Book for printing could also be put into the Book class itself.  Have a look at https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#toString() and see if there's anything there that might help with this.

Reply:
Yes, I can see how that is cleaner. I'll use it for when I break up the MainMenu class.

Reply:
Doing so will also allow you to do most of the testing of the format inside the Book test as well.